extensions/InputBox/InputBox.hooks.php

```diff
@@ -52,24 +52,38 @@ public static function onMediaWikiPerformAction(
 		$request, 
 		$wiki )
 	{
+		global $wgScript, $wgUser;
+
 		if( $wiki->getAction( $request ) !== 'edit' ){
 			# not our problem
 			return true;
 		}
+
+		$params = $request->getValues();
+
+		if ( $request->getText( 'preload', '' ) !== '' ){
+			# Make sure any call with preload uses VE when it matches the user's prefs
+			if ( EditorPreference::isVisualEditorPrimary() && $wgUser->isLoggedIn() ) {
+				$params['veaction'] = 'edit';
+				unset( $params['action'] );
+
+				$output->redirect( wfAppendQuery( $wgScript, $params ), '301' );
+				return false;
```

Comment:
Shouldn't it be this for the early return:

```php
if ( $request->getText( 'prefix', '' ) === '' && $request->getText( 'preload', '' ) === '' ) {
	// Fine
	return true;
}
```

The code sets prefix='' in all requests, so it will always be there, but mostly be empty string.
For preload, the parameter is only present when the InputBox has preload data.

What's the difference between these three?

```php
$request->getText( 'prefix', '' ) === ''
empty( $params['prefix'] )
isset( $params['title'] )
```